pyvista/utilities/reader.py

```diff
@@ -2330,6 +2332,66 @@ class GIFReader(BaseReader):
     _class_reader = _GIFReader
 
 
+class XdmfReader(BaseReader, PointCellDataSelection):
+    """XdmfReader for .xdmf files.
+
+    Parameters
+    ----------
+    path : str
+        Path of the XDMF file to read.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> from pyvista import examples
+    >>> filename = examples.download_dolfinx_mesh(load=False)
+    >>> reader = pyvista.get_reader(filename)
+    >>> filename.split("/")[-1]  # omit the path
+    'mesh.xdmf'
+    >>> mesh = reader.read()
+    >>> mesh.plot()
+
+    """
+
+    _class_reader = staticmethod(_vtk.lazy_vtkXdmfReader)
+
+    @property
+    def number_grids(self):
+        """Get information about grids.
+
+        Returns
+        -------
+        int
+
+        """
+        return self.reader.GetNumberOfGrids()
+
+    @property
+    def number_point_arrays(self):
```

Comment:
This and `point_array_names` look to be unneeded.  Are they different than the methods from `PointCellDataSelection?`

Reply:
You are right, I should have looked more carefully at the PointCellDataSelection class.